reader.js

```diff
@@ -7,15 +9,156 @@ const CLOSE_TIMEOUT = 10000;
 
 let WSS;
 
-function init(wss) {
+function init(server) {
     console.log('Initializing nasonov-reader');
+    const wss = new WebSocket.Server({server});
 
     WSS = wss;
     wss.on('connection', function connected(ws, req) {});
 
+    server.on('request', (request, response) => {connectToInflux(request, response)});
+
     connectToWriter();
 }
 
+function connectToInflux(request, response) {
+
+    if(request.method === 'GET') {
+
+        let requestQuery = new URL(request.url, 'https://habmc.stanfordssi.org/');
+
+        if(requestQuery.pathname.search('/index') > 0) {
+
+            let mission = requestQuery.pathname.substring(1, requestQuery.pathname.search('/index'));
+
+            if (isNaN(mission)) {
+                return;
+            }
+
+            influxConnection.then(function (influx) {
+
```

Comment:
Cool trick to avoid nested promises (which are less readable): chain them using return statements. WTF does that mean?

```
influxConnection.then((influx) => {
    return influx.getMeasurements();
}).then((names) => {
    // your code here
});
```

This also lets you condense it down to one `catch`.